### ISSUE TITLE
Shiny apps don't close

### ISSUE THREAD

**Issue by fvalenduc** [jgenco/jgenco-overlay#20]
If I run a shiny app with rstudio, it does not close when I close the window. This remains in the console:

Attachement du package : ‘lubridate’

Les objets suivants sont masqués depuis ‘package:base’:

    date, intersect, setdiff, union

Listening on http://127.0.0.1:4583

If I click several on the stop button, I finally get this:

Listening on http://127.0.0.1:4583
Called from: top level 
Browse[1]> Q

Called from: execCallbacks(timeoutSecs, all, loop$id)
Browse[1]> Q
> 

Afterwards, the button "run app" is replaced by "reload app" and it is impossible to run the application again without restarting rstudio.
I don't see this problem on opensuse or ubuntu for example.

**Comment by jgenco** [comment]
Trying the Shiny default single file "Old Faithful Geyser Data" script, it closed out fine w/o error with `shiny-1.7.5` and `shiniy-1.8.0` by using `Run App`  with `Run in Window` and `in R Console` checked.

I did discover issue rstudio/shiny#3309, and by setting `Debug -> On Error -> Break in Code`, RStudio produces the console messages you are getting.  But, I'm still able to re-start the Shiny App. Perhaps this is part of the difference?

using rstudio 2023.12+278 electron version:
```
RStudio 2023.12.0+278 "Ocean Storm" Daily (6b9436f529581d79c43abbea8cb08a0f6a3cdd60, 2023-11-11) for Gentoo
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.12.0.278-+278 Chrome/116.0.5845.190 Electron/26.2.4 Safari/537.36
```

**Comment by fvalenduc** [comment]
Thanks for your help. It works correctly with the electron use flag and by setting debug - on error - message only.
Then the shiny app closes cleanly when closing the viewer.
